src/cursor_manager.ts

```diff
@@ -63,9 +64,9 @@ export class CursorManager implements Disposable {
      */
     public wantInsertCursorUpdate = true;
     /**
-     * Set of grid that needs to undergo cursor update
+     * Set of grids that needs to undergo cursor update
      */
-    private gridCursorUpdates: Set<number> = new Set();
+    private gridCursorUpdates: PendingUpdates<number> = new PendingUpdates();
```

Comment:
Nice re-use. This makes me notice that PendingUpdates has set-like behavior, does that mean the `addForceUpdate` method has a misleading name? Because it's not appending something, but rather *setting* a key, which may overwrite the previous entry.

Reply:
Aha! So, yes and no. It has set like behavior in that for each key you add, it only stages that resource as a single entry (check out [this test](https://github.com/vscode-neovim/vscode-neovim/blob/master/src/test/unit/pending_updates.test.ts#L11-L17). If we were to be using conditional updates, this would mean that multiple calls to `addConditionalUpdate` would have one key, with multiple conditions. So no, nothing is overwritten.

As for the name, yeah, I felt weird writing `addForceUpdate` here, since there's no contrasting "conditional" usage. I welcome an alternate name though!